packages/api-plugin-fulfillment-type-shipping/src/startup.js

```diff
@@ -0,0 +1,23 @@
+import ReactionError from "@reactioncommerce/reaction-error";
+import checkAndCreateFulfillmentType from "./checkAndCreateFulfillmentType.js";
+
+/**
+ * @summary Called on startup to create the root entry of this fulfillment type in Fulfillment collection
+ * @param {Object} context Startup context
+ * @param {Object} context.collections Map of MongoDB collections
+ * @returns {undefined}
+ */
+export default async function fulfillmentTypeShippingStartup(context) {
+  context.appEvents.on("afterShopCreate", async (payload) => {
+    const { shop } = payload;
+    const shopId = shop._id;
+
+    // We do not have validatePermissions in context during this startup stage, hence commenting below
+    // await context.validatePermissions("reaction:legacy:fulfillmentTypes", "read", { shopId });
+
+    const insertSuccess = await checkAndCreateFulfillmentType(context, shopId);
```

Comment:
When we create an entity, isn't it implied that we check the conditions anyway? The `checkAnd` part of the function name seems a bit overexplicit.

Reply:
Inside the `checkAndCreateFulfillmentType` we have the call to the original `createFulfillmentType` mutation.

Although the mutation call is explicitly making the call to the one defined in context, I thought I could make this a bit explicit to avoid any quick confusion.